src/segment_evaluation.rs

```diff
@@ -33,19 +33,24 @@ pub(crate) fn find_applicable_segment_rule_for_entity(
     let mut targeting_rules = segment_rules.collect::<Vec<_>>();
     targeting_rules.sort_by(|a, b| a.order.cmp(&b.order));
 
-    for targeting_rule in targeting_rules.into_iter(){
-        if targeting_rule_applies_to_entity(segments, &targeting_rule, entity)
-        .map_err(
-            |e|{
-            // This terminates the use of anyhow in this module, converting all errors:
-            let cause: String = e.chain().map(|c| format!("\nCaused by: {c}")).collect();
-            Error::EntityEvaluationError(format!("Failed to evaluate entity '{}' against targeting rule '{}'.{cause}", entity.get_id(), targeting_rule.order))
-        })?
-        {
-            return Ok(Some(targeting_rule));
-        }
-    }
-    return Ok(None)
+    targeting_rules
+        .into_iter()
+        .find_map(|targeting_rule| {
+            match targeting_rule_applies_to_entity(segments, &targeting_rule, entity) {
+                Ok(true) => Some(Ok(targeting_rule)),
+                Ok(false) => None,
+                Err(e) => {
+                    // This terminates the use of anyhow in this module, converting all errors:
+                    let cause: String = e.chain().map(|c| format!("\nCaused by: {c}")).collect();
+                    Some(Err(Error::EntityEvaluationError(format!(
+                        "Failed to evaluate entity '{}' against targeting rule '{}'.{cause}",
+                        entity.get_id(),
+                        targeting_rule.order
+                    ))))
+                }
+            }
+        })
+        .transpose()
```

Comment:
Following your comment here: https://github.com/IBM/appconfiguration-rust-sdk/pull/21/files#r1873822316, this might be a bit more rusty.

Also, the `targeting_rule_applies_to_entity` call can probably return the error and there is no need to map it here.